ansible/roles/test/tasks/bgp_neighbor/bgp_neighbor_test_ipv4_bounce_peer.yml

```diff
@@ -0,0 +1,49 @@
+# Subtask to bounce an ipv4 neighbor and verify it comes back
+- block:
+  - name: output variables
+    debug:
+      msg: "Exercizing BGP peer link for peer: {{ peer }}"
+
+  - name: shut down neighbor peer links
+    shell: config bgp shutdown neighbor {{ peer }}
+    become: yes
+
+  - name: execute show command
+    shell: show bgp ipv4 summary json
+    register: output
+
+  - name: convert output to json and extract bgp peers
+    set_fact:
+      subtask_bgp_peers: "{{ (output.stdout|from_json).ipv4Unicast.peers }}"
+
+  - name: verify BGP peers are 'Idle (Admin)'
+    assert: { that: "subtask_bgp_peers[peer].state == 'Idle (Admin)'" }
+
+  - name: start up all peer links
```

Comment:
This is just affecting a single BGP neighbor, not "all", so the name is not accurate. Also, saying "peer link" sounds a little ambiguous to me, as if it were a layer 1 or layer 2 connection. I think the term "BGP neighbor" would work better. For example, you can just say "Shut down BGP neighbor"

Reply:
I updated the repo.  some of the variables and comments were from when I had this as a single file.  I fixed the wording in the comments and variable names to reflect 'bgp neighbor' instead of bgp peer and adjusted plurality as needed.